tests/SessionTest.php

```diff
@@ -0,0 +1,121 @@
+<?php
+/*
+-------------------------------------------------------------------------
+GLPI - Gestionnaire Libre de Parc Informatique
+Copyright (C) 2015-2016 Teclib'.
+
+http://glpi-project.org
+
+based on GLPI - Gestionnaire Libre de Parc Informatique
+Copyright (C) 2003-2014 by the INDEPNET Development Team.
+
+-------------------------------------------------------------------------
+
+LICENSE
+
+This file is part of GLPI.
+
+GLPI is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+GLPI is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+--------------------------------------------------------------------------
+*/
+
+/* Test for inc/session.class.php */
+
+class SessionTest extends PHPUnit_Framework_TestCase {
+
+   /**
+    * @covers Session::addMessageAfterRedirect
+    * @covers Html::displayMessageAfterRedirect
+    */
+    public function testAddMessageAfterRedirect() {
+      session_start();
+      $err_msg = 'Something is broken. Weird.';
+      $warn_msg = 'There was a warning. Be carefull.';
+      $info_msg = 'All goes well. Or not... Who knows ;)';
+
+      $this->assertNull($_SESSION['MESSAGE_AFTER_REDIRECT']);
+
+      //test add message in cron mode
+      $_SESSION['glpicronuserrunning'] = 'phpunit_cron';
+      Session::addMessageAfterRedirect($err_msg, false, ERROR);
+
+      $this->assertNull($_SESSION['MESSAGE_AFTER_REDIRECT']);
```

Comment:
Nope, we indeed add a message, but is Session::isCron() is true, anything will be added in the session ;)